functions/DynamicParams.ps1

```diff
@@ -271,6 +271,55 @@ Function Get-ParamSqlLogins
 	return $newparams
 }
 
+Function Get-ParamSqlMaintenancePlans
+{
+<#
+ .SYNOPSIS
+ Internal function. Returns System.Management.Automation.RuntimeDefinedParameterDictionary
+ filled with Maintenance Plans from specified SQL Server Job Server (SQL Agent).
+#>
+	[CmdletBinding()]
+	param (
+		[Parameter(Mandatory = $true)]
+		[Alias("ServerInstance","SqlInstance")]
+		[object]$SqlServer,
+		[System.Management.Automation.PSCredential]$SqlCredential
+	)
+
+	try { $server = Connect-SqlServer -SqlServer $SqlServer -SqlCredential $SqlCredential -ParameterConnection }
+	catch { return }
+
+	$qryPlanNames = "
+	SELECT 
+		sp.[name] AS MaintenancePlans
+	FROM msdb.dbo.sysmaintplan_plans AS sp
+	"
+	$planList = @()
+	$planList = $server.ConnectionContext.ExecuteWithResults($qryPlanNames).Tables.Rows.MaintenancePlans
+	
+	# Reusable parameter setup
+	$newparams = New-Object System.Management.Automation.RuntimeDefinedParameterDictionary
+	$attributes = New-Object System.Management.Automation.ParameterAttribute
+
+	$attributes.ParameterSetName = "__AllParameterSets"
+	$attributes.Mandatory = $false
+	$attributes.Position = 4
+
+	# MaintenancePlan list parameter setup
+	if ($list) { $validationset = New-Object System.Management.Automation.ValidateSetAttribute -ArgumentList $list }
```

Comment:
If you're interested, it was this variable. You used $planList above but $list in this part. Everything else was good.